### PR TITLE
Support CRC32C Checksum for GCS

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?
Support new contentHash for GCS

### Why are the changes needed?
new contenthash so we can compare HDFS checksum with GCS object checksum

### Does this PR introduce any user facing changes?

new property

### CODE REVIEW

**Review by beinan — APPROVED** [review]
lgtm

**Comment by jja725** [comment]
alluxio-bot, merge this please

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -1006,6 +1007,11 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.ENFORCE)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey UNDERFS_CHECKSUM_TYPE =
+      enumBuilder(Name.UNDERFS_CHECKSUM_TYPE, ChecksumType.class)
+          .setDefaultValue(ChecksumType.MD5)
+          .setDescription("UFS checksum type.")
+          .build();
```
Comment (lucyge2022) [inline dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:1014]: do u need to set Scope? does it matter?

Comment (jja725) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:1014]: I think scope is not really used except few checks

**Pull request merged**